rust/core-lib/src/editor.rs

```diff
@@ -900,6 +901,40 @@ impl Editor {
         view.invalidate_styles(&self.text, start, end_offset);
     }
 
+    pub fn update_annotations(
+        &mut self,
+        view: &mut View,
+        plugin: PluginId,
+        start: usize,
+        len: usize,
+        annotation_slices: Vec<AnnotationSlice>,
+        rev: RevToken,
+    ) {
+        let _t = trace_block("Editor::update_annotations", &["core"]);
+
+        for annotation_slice in annotation_slices {
+            let mut start = start;
+            let mut end_offset = start + len;
+            let mut annotations = annotation_slice.to_annotations(view, &self.text);
+
+            if rev != self.engine.get_head_rev_id().token() {
```

Comment:
it's a curious question what we want to do in this case, but this seems reasonable for now!